diff --git a/core/java/android/database/sqlite/SQLiteTokenizer.java b/core/java/android/database/sqlite/SQLiteTokenizer.java
index 39749fb..3edcc07 100644
--- a/core/java/android/database/sqlite/SQLiteTokenizer.java
+++ b/core/java/android/database/sqlite/SQLiteTokenizer.java
@@ -70,6 +70,12 @@
     public static final int OPTION_TOKEN_ONLY = 1 << 0;
 
     /**
+     * Checks brackets; an unbalanced expression or one that attempts to close an
+     * earlier bracket will result in an exception.
+     */
+    public static final int OPTION_CHECK_BRACKETS = 1 << 1;
+
+    /**
      * Tokenize the given SQL, returning the list of each encountered token.
      *
      * @throws IllegalArgumentException if invalid SQL is encountered.
@@ -86,11 +92,13 @@
      *
      * @throws IllegalArgumentException if invalid SQL is encountered.
      */
-    public static void tokenize(@Nullable String sql, int options, Consumer<String> checker) {
+    public static void tokenize(@Nullable String sql, int options,
+            @Nullable Consumer<String> checker) {
         if (sql == null) {
             return;
         }
         int pos = 0;
+        int level = 0;
         final int len = sql.length();
         while (pos < len) {
             final char ch = peek(sql, pos);
@@ -105,7 +113,9 @@
                 final int end = pos;
 
                 final String token = sql.substring(start, end);
-                checker.accept(token);
+                if (checker != null) {
+                    checker.accept(token);
+                }
 
                 continue;
             }
@@ -142,9 +152,11 @@
                     } else {
                         token = tokenUnquoted;
                     }
-                    checker.accept(token);
+                    if (checker != null) {
+                        checker.accept(token);
+                    }
                 } else {
-                    if ((options &= OPTION_TOKEN_ONLY) != 0) {
+                    if ((options & OPTION_TOKEN_ONLY) != 0) {
                         throw genException("Non-token detected", sql);
                     }
                 }
@@ -164,10 +176,12 @@
 
                 final String token = sql.substring(quoteStart + 1, quoteEnd);
 
-                checker.accept(token);
+                if (checker != null) {
+                    checker.accept(token);
+                }
                 continue;
             }
-            if ((options &= OPTION_TOKEN_ONLY) != 0) {
+            if ((options & OPTION_TOKEN_ONLY) != 0) {
                 throw genException("Non-token detected", sql);
             }
 
@@ -199,11 +213,31 @@
                 throw genException("Semicolon is not allowed", sql);
             }
 
+            if (ch == '(') {
+                pos++;
+                level++;
+
+                continue;
+            }
+
+            if (ch == ')') {
+                pos++;
+                level--;
+                if (level < 0 && (options & OPTION_CHECK_BRACKETS) != 0) {
+                    throw genException("Unbalanced brackets", sql);
+                }
+
+                continue;
+            }
             // For this purpose, we can simply ignore other characters.
             // (Note it doesn't handle the X'' literal properly and reports this X as a token,
             // but that should be fine...)
             pos++;
         }
+
+        if (level != 0 && (options & OPTION_CHECK_BRACKETS) != 0) {
+            throw genException("Unbalanced brackets", sql);
+        }
     }
 
     /**
